scipy/stats/_axis_nan_policy.py

```diff
@@ -493,21 +493,22 @@ def hypotest_fun_out(*samples, **kwds):
             ndims = np.array([sample.ndim for sample in samples])
             if np.all(ndims <= 1):
                 # Addresses nan_policy == "raise"
-                contains_nans = []
-                for sample in samples:
-                    contains_nan, _ = _contains_nan(sample, nan_policy)
-                    contains_nans.append(contains_nan)
+                if nan_policy != 'propagate' or override['nan_propagation']:
+                    contains_nan = [_contains_nan(sample, nan_policy)[0]
+                                    for sample in samples]
+                else:
+                    # Behave as though there are no NaNs (even if there are)
+                    contains_nan = [False]*len(samples)
 
                 # Addresses nan_policy == "propagate"
-                if any(contains_nans) and (nan_policy == 'propagate'
-                                           and override['nan_propagation']):
+                if any(contains_nan) and nan_policy == 'propagate':
```

Comment:
```suggestion
                if any(contains_nans) and (nan_policy == 'propagate'
                                           and override['nan_propagation']):
```
Not necessary, but might as well leave this alone to be explicit.

Reply:
Oops `contains_nans` needed to be `contains_nan`.